sky/skylet/providers/lambda_labs/lambda_utils.py

```diff
@@ -0,0 +1,160 @@
+
+        # Optimization:
+        # Most API requests are rate limited at ~1 request every second but
+        # launch requests are rate limited at ~1 request every 10 seconds.
+        # So don't use launch requests to check availability.
+        # See https://docs.lambdalabs.com/cloud/rate-limiting/ for more.
+        available_regions = self.ls_catalog()['data'][instance_type]\
+                ['regions_with_capacity_available']
+        available_regions = [reg['name'] for reg in available_regions]
+        if region not in available_regions:
+            if len(available_regions) > 0:
+                aval_reg = ' '.join(available_regions)
+            else:
+                aval_reg = 'None'
+            raise LambdaLabsError(('instance-operations/launch/'
+                                   'insufficient-capacity: Not enough '
+                                   'capacity to fulfill launch request. '
+                                   'Regions with capacity available: '
+                                   f'{aval_reg}'))
+
+        # Try to launch instance
+        data = json.dumps({
+                    'region_name': region,
+                    'instance_type_name': instance_type,
+                    'ssh_key_names': [
+                        self.ssh_key_name
+                    ],
+                    'quantity': quantity,
+                    'name': name
+                })
+        response = requests.post(f'{API_ENDPOINT}/instance-operations/launch',
+                                 data=data,
+                                 headers=self.headers)
+        raise_lambda_error(response)
+        return response.json()
+
+    def rm(self, *instance_ids):
+        """Terminate instances."""
+        data = json.dumps({
+            'instance_ids': [
+                instance_ids[0] # TODO(ewzeng) don't hardcode
+            ]
+        })
+        response = requests.post(f'{API_ENDPOINT}/instance-operations/terminate',
+                                 data=data,
+                                 headers=self.headers)
+        raise_lambda_error(response)
+        return response.json()
+
+    def ls(self):
+        """List existing instances."""
+        response = requests.get(f'{API_ENDPOINT}/instances', headers=self.headers)
+        raise_lambda_error(response)
+        return response.json()
+
+    def set_ssh_key(self, name, pub_key):
+        """Set ssh key."""
+        data = json.dumps({
+            'name': name,
+            'public_key': pub_key
+        })
+        response = requests.post(f'{API_ENDPOINT}/ssh-keys',
+                                 data=data,
+                                 headers=self.headers)
+        raise_lambda_error(response)
+        self.ssh_key_name = name
+        with open(self.credentials, 'w') as f:
+            f.write(f'api_key={self.api_key}\n')
+            f.write(f'ssh_key_name={self.ssh_key_name}\n')
+
+    def ls_catalog(self):
```

Comment:
Is this needed? Does the CSV not provide this?

Reply:
Good question! This function is different from the csv because it only lists regions that have availability (i.e. not out of capacity).